src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -4420,23 +4420,15 @@ protected RubyNode coercePatternToRope(RubyNode pattern) {
         @Specialization(
                 guards = {
                         "offset >= 0",
-                        "singleByteOptimizableNode.execute(stringRope)",
-                        "!patternFits(stringRope, patternRope, offset)" })
-        protected Object patternTooLarge(Rope stringRope, Rope patternRope, int offset) {
-            return nil;
-        }
-
-        @Specialization(
-                guards = {
-                        "offset >= 0",
-                        "singleByteOptimizableNode.execute(stringRope)",
-                        "patternFits(stringRope, patternRope, offset)" })
+                        "singleByteOptimizableNode.execute(stringRope)" })
         protected Object singleByteOptimizable(Rope stringRope, Rope patternRope, int offset,
                 @Cached RopeNodes.BytesNode stringBytesNode,
                 @Cached RopeNodes.BytesNode patternBytesNode,
                 @Cached LoopConditionProfile loopProfile,
                 @Cached("createCountingProfile()") ConditionProfile matchProfile) {
 
+            assert offset + patternRope.byteLength() <= stringRope.byteLength(); // otherwise returns in ruby fastpath
```

Comment:
The existing TruffleRuby code seems to [prefer](https://github.com/oracle/truffleruby/blob/vm-21.1.0/src/main/java/org/truffleruby/parser/ast/ModuleParseNode.java#L49) giving `assert` a detail message argument rather than a comment, so it might be worth doing that here & below.